addons/mail/static/src/core/common/autoresize_input.xml

```diff
@@ -4,13 +4,14 @@
 <t t-name="mail.AutoresizeInput">
     <input
         class="o-mail-AutoresizeInput o_input px-1 border-1 text-truncate"
-        t-attf-class="{{ props.className }}"
+        t-attf-class="{{ props.className }} {{ state.isFocused ? 'o-focused' : '' }}"
```

Comment:
Nitpicking: conditional static classnames are best with `t-att-class` object syntax.
I would define `class`, `t-attf-class` and `t-att-class` (object syntax) and let OWL combine them nicely.

I usually tend to reserve `t-attf-class` for `{{ props.className }}` when a component become complex with classnames, and `t-att-class` usually becomes object syntax for relatively best maintainability when lots of conditional classnames.

Reply:
`props.className` could also be used with `t-att-class` object syntax with `{ [props.className]: true }`...
Too many choices!

That's honestly very subjective with current use of dynamic classnames, so you can really ignore this comment because that's not really worth caring about.